### PR TITLE
Added application layer factory reset(offline)

### LINKED ISSUE #304: To have a factory reset mode on an application level
**Is your feature request related to a problem? Please describe.**
Currently, the factory reset mode requires an internet connection to clone and reset the repo

**Describe the solution you'd like**
A backup repo/complete SUSI Linux file to be stored e.g. as a zip and reinstalled in case of failure or wish by the user to reset
Following directories should be present in the backup:
1>SUSI Linux
2>SUSI Server
3>Skill Data
Other: Zip files must be installed during the installation process (e.g. copy these 3 items as a zip on the device and decompress them. Then install the device from there)
Have a tar file of the local susi_server installation to overwrite a failed update

**Additional context**
I would like to work on this issue

### PR DESCRIPTION
**Pull request by stealthanthrax**
Fixes #304 

Now a backup partition is kept in the repo for offline factory reset

### CODE REVIEW

File: factory_reset/factory_reset.sh [hunk 1]
```diff
@@ -4,14 +4,12 @@
...
-mv susi_linux/ susi_temp
-git clone https://github.com/fossasia/susi_linux #while testing change to personal repo
-pwd
-ls
-cd susi_linux
 
-rm -rf ../susi_temp
+sudo tar xvf $DIR_PATH/.backup.tar.gz  # extracting the backup script
+mv $DIR_PATH/../susi_linux $DIR_PATH/../../susi_temp  # moving the backup file as a temp file
+mv $DIR_PATH/../../susi_linux $DIR_PATH/../../susi_temp2  # making the original file to a temp file to delete it
```
Comment (prateekiiest) [inline factory_reset/factory_reset.sh]: Looks good

**Comment by stealthanthrax** [comment]
@prateekiiest , have a look too

**Comment by mariobehling** [comment]
Thanks. Please resolve conflicts.

**Comment by stealthanthrax** [comment]
@mariobehling @cweitat , I have resolved the conflicts

**Pull request merged**